datafusion/expr/src/logical_plan/ddl.rs

```diff
@@ -284,6 +346,15 @@ pub struct CreateCatalogSchema {
     pub schema: DFSchemaRef,
 }
 
+impl PartialOrd for CreateCatalogSchema {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        match self.schema_name.partial_cmp(&other.schema_name) {
```

Comment:
Perhaps you can add a comment explaining it is due to not comparing schema ref

Another way you could do this might be with some inner struct like:

```rust
impl PartialOrd for CreateCatalogSchema {
    fn partial_cmp(&self, other: &Self) -> Option<Ordering> { 
      #[derive(PartialOrd)]
      struct ComparableCreateCatalogSchema<'a> {
       /// The table schema
       pub schema_name: &'astr,
      }
      let comparable_self = ComparableCreateCatalogSchema { schema_name: &self.schema_name}
      let comparable_other= ComparableCreateCatalogSchema { schema_name: &other.schema_name}
      comparable_self.partial_cmp(comparable_other)
   }
}
```

For this particular impl it probably doesn't matter but for `CreateExternalTable` it might be easier to follow and verify it is correct

Reply:
I like this, I'll make some changes for the bigger manual implementations to follow this format to begin with!